mongo/options/searchindexoptions.go

```diff
@@ -22,6 +23,12 @@ func (sio *SearchIndexesOptions) SetName(name string) *SearchIndexesOptions {
 	return sio
 }
 
+// SetType sets the value for the Type field.
+func (sio *SearchIndexesOptions) SetType(ty string) *SearchIndexesOptions {
+	sio.Type = &ty
```

Comment:
Optional. Suggest using `typ` over `ty` as a parameter name for setting `Type`

```suggestion
func (sio *SearchIndexesOptions) SetType(typ string) *SearchIndexesOptions {
	sio.Type = &typ
```